python/cugraph/structure/symmetrize.py

```diff
@@ -75,28 +72,84 @@ def symmetrize_df(df, src_name, dst_name):
     return gdf.groupby(by=[src_name, dst_name], as_index=False).min()
 
 
+def symmetrize_ddf(df, src_name, dst_name, weight_name=None):
+    """
+    Take a COO stored in a distributed DataFrame, and the column names of
+    the source and destination columns and create a new data frame
+    using the same column names that symmetrize the graph so that all
+    edges appear in both directions.
+
+    Note that if other columns exist in the data frame (e.g. edge weights)
+    the other columns will also be replicated.  That is, if (u,v,data)
+    represents the source value (u), destination value (v) and some
+    set of other columns (data) in the input data, then the output
+    data will contain both (u,v,data) and (v,u,data) with matching
+    data.
+
+    If (u,v,data1) and (v,u,data2) exist in the input data where data1
+    != data2 then this code will arbitrarily pick the smaller data
+    element to keep, if this is not desired then the caller should
+    should correct the data prior to calling symmetrize.
+
+    Parameters
+    ----------
+    df : dask_cudf.DataFrame
+        Input data frame containing COO.  Columns should contain source
+        ids, destination ids and any properties associated with the
+        edges.
+    src_name : string
+        Name of the column in the data frame containing the source ids
+    dst_name : string
+        Name of the column in the data frame containing the destination ids
+
+    Examples
+    --------
+    >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> sym_df = cugraph.symmetrize(M, '0', '1')
+    >>> G = cugraph.Graph()
+    >>> G.add_edge_list(sym_df['0]', sym_df['1'], sym_df['2'])
+    """
+    if weight_name:
+        ddf2 = df[[dst_name, src_name, weight_name]]
+        ddf2.columns = [src_name, dst_name, weight_name]
+    else:
+        ddf2 = df[[dst_name, src_name]]
+        ddf2.columns = [src_name, dst_name]
+
+    ddf = df.append(ddf2).reset_index(drop=True)
+    result = (
+        ddf.groupby(by=[src_name, dst_name], as_index=False)
+        .min()
```

Comment:
This is the point where we break a MultiGraph...

Reply:
Right, cugraph doesn't support MG multigraph or SG multigraph in symmetrize atm.